extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisReceiver.scala

```diff
@@ -68,7 +64,8 @@ private[kinesis] class KinesisReceiver(
     endpointUrl: String,
     checkpointInterval: Duration,
     initialPositionInStream: InitialPositionInStream,
-    storageLevel: StorageLevel)
+    storageLevel: StorageLevel,
+    credentials: Option[AWSCredentials])
```

Comment:
I am really not sure how this will work. AWSCredentials is not serializable, while the Receiver has to be serializable (as the receiver object is sent to the executors from the driver). Did you test whether this works?